### PR TITLE
Add Element@1.0.0-rc.7 w/ npm auto-update

### PR DESCRIPTION
**Pull request by lizheming**
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Profile of the lib
- Git repository (required): https://github.com/ElemeFE/element
- Official website (optional, not the repository): http://element.eleme.io
- NPM package url (optional): https://www.npmjs.com/package/element-ui
- GitHub / Bitbucket popularity (required): 
  - Count of stars: 3735
  - Count of watchers: 178
  - Count of forks: 403
- NPM download stats (optional):
  - Downloads in the last day: 140
  - Downloads in the last week: 879
  - Downloads in the last month: 3239
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [ ] No already exist / duplicated issue and PR
- [x] The lib has notable popularity
  - [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [x] More than 500 downloads stats per month on npm registry
- [x] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [ ] Has valid tags for each versions (for git auto-update)
- [ ] Auto-update setup
- [ ] Auto-update target/source is valid.
- [ ] Auto-update filemap is correct.
# Git commit checklist
- [ ] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [ ] The parent of the commit(s) in the PR is not old than 3 days.
- [ ] Pull request is sending from a non-master branch with meaningful name.
- [ ] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [ ] Close corresponding issue in commit message
- [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/element-ui/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "license": "MIT",
+  "npmName": "element-ui",
+  "npmFileMap": [
+    {
+      "basePath": "/lib",
+      "files": [
+	"index.js",
+        "element-ui.common.js",
+        "theme-default/index.css",
+        "theme-default/fonts/*"
```
Comment (x09326) [inline ajax/libs/element-ui/package.json]: @lizheming 
There is a space in the line of `index.js`.
Please remove it.

File: ajax/libs/element-ui/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  "version": "1.0.0-rc.7",
+  "description": "A Component Library for Vue.js.",
+  "main": "index.js",
+  "filename": "index.js",
+  "files": [
+    "index.js",
+    "element-ui.common.js",
+    "theme-default/index.css",
+    "theme-default/fonts/"
+  ],
```
Comment (x09326) [inline ajax/libs/element-ui/package.json]: I think we don't need this `files` field.
Please remove it.

File: ajax/libs/element-ui/package.json (hunk 1 above)
Comment (lizheming) [inline ajax/libs/element-ui/package.json]: Removed, please check again, thank you!

File: ajax/libs/element-ui/package.json (hunk 2 above)
Comment (lizheming) [inline ajax/libs/element-ui/package.json]: Removed, thank you !

**Comment by x09326** [comment]
@lizheming 
Please rebase with the latest branch.

**Comment by lizheming** [comment]
@x09326 Where commit should I mention author, in this pr or my fork commit? Does mention them to confirm files correction in package or anything else? Thank you!

**Comment by x09326** [comment]
@lizheming 
The commit in this PR.
Just cc to him/her then he or she will know the lib has been hosted on cdnjs.

**Comment by QingWei-Li** [comment]
I am one of the authors. Thanks @lizheming 
What do I need to do?

**Comment by lizheming** [comment]
@QingWei-Li Thank you. Can you confirm those files in  https://github.com/cdnjs/cdnjs/pull/9345/files are required when people use? And is there anything I lost?

**Comment by QingWei-Li** [comment]
@lizheming 

LGTM.

I have one question. will this be automatically updated from the git?

**Comment by lizheming** [comment]
@QingWei-Li Not yet now. cdnjs support auto update from git and npm.  Element seems use compiled file after npm publish and  repo's release is source file, not compiled file, so cdnjs can't update from git. And npm's version have some problem, it seems we must use `npm install element-ui@latest` to download the right version. So I have not set auto update from npm.

**Comment by QingWei-Li** [comment]
@lizheming I see, thanks.

**Comment by lizheming** [comment]
@QingWei-Li you 2 welcome ~

**Comment by QingWei-Li** [comment]
@lizheming If I put `lib` under another repo, it can automatically updated from the repo? likes https://github.com/ElementUI/lib

**Comment by lizheming** [comment]
@QingWei-Li I think you just need add compiled version when git release. @x09326 Does it right?

**Comment by PeterDaveHello** [comment]
Hi @lizheming @QingWei-Li, thanks for being interested in CDNJS, I'm the maintainer of CDNJS, sorry for the delay. Actually, npm looks fine for me here, since we only fetch the tarball and extract the files in it, not the `npm install` process, and we currently prefer to use npm auto-update until we refactor the git auto-updater for the performance issue.

I just wonder:

1. If it's possible to give the file a better naming scheme? Rather than `tooltip/index.js` maybe just `tooptip.js`? 
2. If you'll like to provide minified files for better performance on CDN? Although CDNJS will still do it if upstream doesn't do that, the official built is preferred.

@QingWei-Li What do you think?

File: ajax/libs/element-ui/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "element-ui",
+  "version": "1.0.0-rc.7",
+  "description": "A Component Library for Vue.js.",
+  "main": "index.js",
```
Comment (pvnr0082t) [inline ajax/libs/element-ui/package.json]: @lizheming Please remove `main` field. Thanks.

**Comment by QingWei-Li** [comment]
Just need `lib/index.js`, `lib/theme-default/index.css` and font files

File: ajax/libs/element-ui/package.json (hunk 3 above)
Comment (lizheming) [inline ajax/libs/element-ui/package.json]: removed, thx!

**Comment by QingWei-Li** [comment]
Please update the author to `ElemeFE`, thanks : P @lizheming

**Comment by lizheming** [comment]
@PeterDaveHello Sorry for response too late. The package Element is special, it seems being refactor, npm download directly may not the right version. At this moment Let me update manually, after they refactored in the future I will modify auto-update setting. and your advice is so good, so I remove main key in package.json. At the last, minified files looks great, thank you!

**Comment by lizheming** [comment]
@QingWei-Li OK, I removed unused file and changed author to ElemeFE, thank you!

**Comment by PeterDaveHello** [comment]
@lizheming as I mentioned, we won't face the problem you said, so no worry :) Would you please also minify these js and css files?

@QingWei-Li any change to provide officially minified assets?

Thanks.

**Comment by lizheming** [comment]
@PeterDaveHello OK，I will see auto-update version and check them. And I found the compiled js and css in commit was already minified, so should I rename them to `index.min.js` and  `index.min.css` ?

**Comment by QingWei-Li** [comment]
@lizheming  We only provide minified umd js file and css file, so do not need to change the name.

**Comment by PeterDaveHello** [comment]
@lizheming @QingWei-Li yes I found that they are already minified, so no need to rename it, thanks!

**Comment by lizheming** [comment]
@PeterDaveHello OK, thank you very much!

**Comment by PeterDaveHello** [comment]
@lizheming you're welcome, thanks for your contribution!

**Comment by PeterDaveHello** [comment]
PR merged! Thanks @lizheming @QingWei-Li @pvnr0082t @x09326 ! Element-ui will be available on CDNJS.com very soon

File: ajax/libs/element-ui/package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    "components"
+  ],
+  "license": "MIT",
+  "npmName": "element-ui",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "index.js",
+        "theme-default/+(index.css|fonts/*)"
```
Comment (PeterDaveHello) [inline ajax/libs/element-ui/package.json]: Looks like this rule could not work, I'll patch this.

**Comment by lizheming** [comment]
Thank you for @PeterDaveHello @pvnr0082t @x09326 's work, and thanks ElemeFE for contribution in Element project.

File: ajax/libs/element-ui/package.json (hunk 4 above)
Comment (lizheming) [inline ajax/libs/element-ui/package.json]: OK, thank you. I add project first for cdnjs, so some rule I may mistake. If you find some error, please help me correct them.

Comment (PeterDaveHello) [inline ajax/libs/element-ui/package.json]: No problem @lizheming

**Comment by PeterDaveHello** [comment]
https://cdnjs.com/libraries/element-ui :100:

**Comment by QingWei-Li** [comment]
Cool! Thank you everybody!

**Comment by lizheming** [comment]
And https://cdn.baomitu.com/element-ui is also well done! 💯

**Comment by QingWei-Li** [comment]
@lizheming Awesome! Thank you!

**Pull request merged**